internal/fs/inode/dir.go

```diff
@@ -939,3 +942,17 @@ func (d *dirInode) ShouldInvalidateKernelListCache(ttl time.Duration) bool {
 	cachedDuration := d.cacheClock.Now().Sub(d.prevDirListingTimeStamp)
 	return cachedDuration >= ttl
 }
+
+func (d *dirInode) RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (folder *gcs.Folder, err error) {
+	folder, err = d.bucket.RenameFolder(ctx, folderName, destinationFolderId)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove old entry from type cache.
+	d.cache.Erase(folderName)
+	// Add new renamed folder in type cache.
+	d.cache.Insert(d.cacheClock.Now(), destinationFolderId, metadata.ExplicitDirType)
```

Comment:
@vadlakondaswetha should we add a TODO here, cache update will not needed once type cache usage is removed from hns

Reply:
Done